### PR TITLE
Test proccess on chainstart

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #3409

adding test to #3486

### CODE REVIEW

File: beacon-chain/sync/subscriber.go [hunk 1]
```diff
@@ -15,6 +16,8 @@ import (
 
 const oneYear = 365 * 24 * time.Hour
 
+var f *event.Feed
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go]: why is this put here ?

File: beacon-chain/sync/subscriber.go [hunk 2]
```diff
@@ -41,9 +44,9 @@ func noopValidator(_ context.Context, _ proto.Message, _ p2p.Broadcaster, _ bool
 func (r *RegularSync) registerSubscribers() {
 	go func() {
 		ch := make(chan time.Time)
-		sub := r.chain.StateInitializedFeed().Subscribe(ch)
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go]: why is it changed

File: beacon-chain/sync/subscriber_test.go [hunk 3]
```diff
@@ -2,23 +2,40 @@ package sync
...
 	"time"
 
 	"github.com/gogo/protobuf/proto"
+	mockChain "github.com/prysmaticlabs/prysm/beacon-chain/blockchain/testing"
 	p2ptest "github.com/prysmaticlabs/prysm/beacon-chain/p2p/testing"
 	pb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
 	"github.com/prysmaticlabs/prysm/shared/testutil"
 )
 
+func checkPanic(want error, fn func()) (err error) {
```
Comment (nisdas) [inline beacon-chain/sync/subscriber_test.go]: this isnt used anywhere

File: beacon-chain/sync/subscriber_test.go [hunk 4]
```diff
@@ -40,6 +57,38 @@ func TestSubscribe_ReceivesValidMessage(t *testing.T) {
...
 }
 
+func TestSubscribe_WaitToSync(t *testing.T) {
+	p2p := p2ptest.NewTestP2P(t)
+	r := RegularSync{
+		ctx: context.Background(),
+		p2p: p2p,
+	}
+
+	r.chain = &mockChain.ChainService{}
```
Comment (nisdas) [inline beacon-chain/sync/subscriber_test.go]: if you need to get the test to work you can modify `mockChain.ChainService{}` definition. Allow it to define a state initialized feed and then use that everytime
```go
// ChainService defines the mock interface for testing
type ChainService struct {
	State               *pb.BeaconState
	Root                []byte
	FinalizedCheckPoint *ethpb.Checkpoint
	StateFeed           *event.Feed
}

// StateInitializedFeed mocks the same method in the chain service.
func (ms *ChainService) StateInitializedFeed() *event.Feed {
	if ms.StateFeed != nil {
		return ms.StateFeed
	}
	return new(event.Feed)
}
```

Comment (shayzluf) [reply beacon-chain/sync/subscriber_test.go]: checking if it doesn't break other tests

File: beacon-chain/sync/subscriber_test.go [hunk 5]
```diff
@@ -2,11 +2,14 @@ package sync
 
 import (
 	"context"
+	"github.com/prysmaticlabs/prysm/shared/bls"
```
Comment (nisdas) [inline beacon-chain/sync/subscriber_test.go]: run goimports

**Pull request merged**